packages/browser/src/core/events/index.ts

```diff
@@ -235,21 +292,18 @@ export class EventFactory {
     const [context, overrides] = this.context(event)
     const { options, ...rest } = event
 
-    const body = {
+    const newEvent: SegmentEvent = {
       timestamp: new Date(),
       ...rest,
       context,
       integrations: allIntegrations,
       ...overrides,
     }
+    // NOTE() This is a solution that was implemented a long time ago to prevent some hash collision issue.
```

Comment:
Slight opportunistic refactor -- this is one of the most confusing files in the code base.